### PR TITLE
chore: Bump development mode detector

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -182,7 +182,7 @@
     "@vaadin/time-picker": "24.1.0-alpha3",
     "@vaadin/tooltip": "24.1.0-alpha3",
     "@vaadin/upload": "24.1.0-alpha3",
-    "@vaadin/vaadin-development-mode-detector": "2.0.5",
+    "@vaadin/vaadin-development-mode-detector": "2.0.6",
```
Comment (ZheSun88) [inline package.json:185]: Can we use it as the same way in `DevDependency`?https://github.com/vaadin/bundles/blob/73ed09effdddf830f67850a520e1bae61e69c41d/package.json#L106

so we dont need to update it manually any more..

Comment (Artur-) [reply package.json:185]: It needs to match what is used elsewhere. Why are all other dependencies specified as a given version?

Comment (ZheSun88) [reply package.json:185]: @platosha probably you know more about this..

Comment (Artur-) [reply package.json:185]: No harm in merging it like this though

Comment (ZheSun88) [reply package.json:185]: true that...

but now we have another issue, bundles version is following Webcomponents release. so we need WC 24.0.1 for the 24.0 branch to pick this change (the Cherry Picked one)

**Review by ZheSun88 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (platosha) [reply package.json:185]: Yes, `peerDependencies` are exact versions with the ones bundled in this package, to indicate for `npm` the versions that are expected in the user's application.

**Pull request merged**